### DISCUSSION TITLE
Is there a good documentation for websocket / control client usage?

### DISCUSSION THREAD

**Discussion by ecrode** [snapcast/snapcast#1254] [Q&A]
I would like to retire my hifiberry setup for a VM solution using a USB dac into an older Denon AVR with serial control interface.
In the long term, I'm planning on implementing a multiroom solution not via multitude of clients / amps, but rather by getting info on which zone should be ON, what is the volume for said zone and controlling Denon via custom code (more than happy to share it once I have a working version). In my instance I'm looking only to selectively enable zones for a same stream (no need for different music in different rooms, at least in my case).
In any case Snapcast seems to be the perfect solution for my needs, while not out of the box it seems to support airplay and spotify audio input, a linux based audio client via USB dac and websocket access from "control" clients. I would like to add a new client that would connect and monitor the sentral server to know and control Denon's power, and individual zone volume levels based on the user selection.
Can anyone point me to the documentation as to how the control client should establish connections and ask for ongoing status updates (is anything playing? volume(s), etc...)

Thank you,
Gene

**Comment by LeoSum8** [comment]
Sounds like something you might be able to achieve with home assistant. It can monitor/control snapcast and also demon receivers.
You might be able to get the desired functionality via automations within Home Assistant.
